### DISCUSSION TITLE
Headless Wordpress via Graphql is too slow

### DISCUSSION THREAD

**Discussion by predaytor** [planetscale/discussion#463] [Q&A] [answered]
I successfully migrated wordpress mariadb (size about 5MB) to planetscale (though with some errors in wp console). Testing on local project.

Overall wordpress response speed (as well as sending graphql queries) is too slow, on large sets about 20-25 seconds.

The planetscale database server is located in Frankfurt, I think the latency should be good (I'm from Ukraine). What could be the problem?

![wordpress graphiql with network tab open](https://user-images.githubusercontent.com/32335730/236922192-92da2cc2-8fce-44c0-b6db-39e6d6601808.png)

**Comment by mscoutermarsh** [comment]
Hi @predaytor,

Could you take a look at the Insights page to see what the query speed looks like?

That will help determine where the problem is: are the queries slow? If not: then it makes sense to look into network latency.

**Reply by predaytor** [reply]
Running a complex graphql query via Insomnia to the `localhost/wp/graphql` endpoint (took about 14 seconds).

The number of edges is limited to 100, but there are currently only 18 nodes (projects) in CMS.

```gql
query Projects($first: Int! = 100, $before: String, $after: String) {
  projects(
    first: $first
    where: {orderby: {field: DATE, order: DESC}}
    before: $before
    after: $after
  ) {
    edges {
      cursor
      node {
        id
        slug
        title
        coverGallery {
          id
        }
      }
    }
    pageInfo {
      total
      hasPreviousPage
      hasNextPage
      startCursor
      endCursor
    }
  }
}
```

![Planetscale Insights log](https://user-images.githubusercontent.com/32335730/237058446-2e3a197e-2383-41e7-9bfc-15f2f936a8a4.jpeg)

**Reply by predaytor** [reply]
Simple queries (single project post type) take about 3 seconds. I will try to deploy my application in the same location (Frankfurt) to check the latency in production.

**Reply by mscoutermarsh** [reply]
Great, let me know how it goes!

From the screenshot, it looks like the queries are all fast. Network latency could be the problem.

**Reply by predaytor** [reply]
I redeployed the application in London (fly.io) and planetscale database to `AWS eu-west-2`. In the first test, a few days ago, the latency was very inconsistent (total query time around 40 seconds), and the query latency matched the Insights tab latency (`total time` summary in each row).

Today's testing shows the same (30-40s) in query response on production website, but the Insights tab summary looks good.

![planetscale insights tab (query latency)](https://github.com/planetscale/discussion/assets/32335730/67589463-fcac-424c-8eab-b9775c88f66a)

![planetscale (devtools network tab) response timing](https://github.com/planetscale/discussion/assets/32335730/c6caeeff-2be5-4e91-a134-f5e18a336d73)

I copied the same database from MariaDB dump to Google Cloud MySQL located at `europe-west2-c` (MySQL 8.0.26). The problem is gone, it works pretty fast with every request (no caching).
|     |      |
|-----|------|
| vCPUs | 1 |
| Memory | 3.75 GB |
| SSD storage | 10 GB |

![Google Cloud MySQL](https://github.com/planetscale/discussion/assets/32335730/864b727e-3a9b-4eca-b9a6-382f20653fa0)

I assume that the WPGraphql extension could generate query schema (whole wordpress) every time a query is run using Planetscale. Maybe I'll try to test things on a clean database without an imported dump, both REST API and Graphql. Not sure what is causing the problem.

**Reply by mscoutermarsh** [reply, marked as answer]
Could you try swapping your connection string from `aws.connect.psdb.cloud` to `aws-eu-west-2.connect.psdb.cloud`?

The request might not be getting routed efficiently.

We saw a [similar issue recently between Fly and Us](https://community.fly.io/t/planetscale-large-latency/12576/8?u=kurt) in that region.

**Reply by predaytor** [reply]
Thank you, it works!
